xtrack/beam_elements/elements_src/lineartransfermatrix.h

```diff
@@ -61,12 +67,15 @@ void LinearTransferMatrix_track_local_particle(LinearTransferMatrixData el, Loca
     double new_px = LocalParticle_get_px(part);
     double new_py = LocalParticle_get_py(part);
     double delta = LocalParticle_get_delta(part);
+    double zeta_no_disp = LocalParticle_get_zeta(part);
 
     // removing dispersion and close orbit
     new_x -= disp_x_0 * delta + x_ref_0;
-    new_px -= px_ref_0;
+    new_px -= disp_px_0 * delta + px_ref_0;
```

Comment:
delta and zeta are not canonically conjugate. Using a symplectic matrix with these coordinates results in a non-symplectic map